src/cluster_legacy.c

```diff
@@ -2731,6 +2731,8 @@ int nodeUpdateAddressIfNeeded(clusterNode *node, clusterLink *link, clusterMsg *
 void clusterSetNodeAsPrimary(clusterNode *n) {
```

Comment:
There are other places we replicationUnsetPrimary after clusterSetNodeAsPrimary. I suppose if the expectation is that we just need it here, I think we can remove the other places we call it.